bench_data/glorious_old_parser

```diff
@@ -0,0 +1,8562 @@
+use crate::ast::{AngleBracketedArgs, ParenthesizedArgs, AttrStyle, BareFnTy};
```

Comment:
I am a little bit confused here, so is it "synthetic generation" ? :)
The problem I faced in mbe case was similar : It is quite hard to generate test cases which is good enough for benchmarking ..

Reply:
Yeah, not quite absolutely synthetic indeed. But it feels about right -- it's
not that big, it is self-contained, and it reasonably exercises the parser
infra.

I wonder if we can do something like this for macro expansion? A nice thing
about mbe is that it is much more uniform than parsing (there are three kinds of
repetitions and a handful of meta vars kinds, as opposed to 100+ syntactic
constructs).

What if we pick a dozen of macros that take the longest time to expand? Or the
longest relative to the size of input/output time? I think it should be possible
to get a representative fixture which is ~100kb as opposer to 6mb.

As a separate aspect, I think what really kills us are macros which are
recursive. It would be nice to also add a more integrated expansion benchmark in
the `hir_def` layer. Computing a def map, where we need to resolve few imports,
but to expand many macros, might do the trick.

So yeah, I think the ideal macro bench would be a single 100kb file, which
contains both macro definitions and macro invocation. For a unit benchmark, we
manually extract invocations, and bench "single-step" macro expansion in the
`mbe` crate. Then theres' a separate bench in the `hir_def` which builds a def
map for this file/crate.

To make this fixtere file, I'd run analysis-stats, and looked for extreme macros
-- biggest input, biggest output, biggest "blow up", largest number of arms,
largest depth. Might also make sense to check rusct benchmarks out.

Reply:
Yeah, that's resolved my confusion and I am clearer now, thank you :heart:

Reply:
To clarify, I am myself not sure where I want to end up with our benchmark! I do want to do something unconventional with:

* using #[test]
* not using real projects
* not using std

I hope that such unit-benchmarks will be much less friction to use. Our current approachs with cli bench and analysis stats require to much setup imo.

Reply:
>extreme macros

BTW, this is how `glorious_old_parser` came into existence -- it is the biggest non-generated Rust file there was.